consent-ui/src/app/common/enter-tan/enter-tan.component.ts

```diff
@@ -33,12 +37,28 @@ export class EnterTanComponent implements OnInit {
     this.reportScaResultForm = this.formBuilder.group({
       tan: ['', Validators.required]
     });
-    this.tanConfig = {
-      type: TanType.PIN,
-      data: '17850120452019980412345678041234567804123456789E',
-      description:
-        'We have sent you the confirmation number. Please check your ' + this.scaType + ' and fill in the field below.'
-    };
+
+    this.consentAuthorizationService
+      .authUsingGET(this.authorizationSessionId, this.redirectCode, 'response')
+      .subscribe(response => {
+        let message = 'Please check your ' + this.scaType + ' and fill in the field below.';
+        this.sessionService.setRedirectCode(
+          this.authorizationSessionId,
+          response.headers.get(ApiHeaders.REDIRECT_CODE)
+        );
+
+        const tmp: any = response.body;
+        const challengeData = tmp.challengeData;
+
+        if (challengeData.data[0] != null) {
```

Comment:
I think this condition is not stable - it ignores `this.scaType` that is the primary source of truth. Correct condition should look like:
```
if ( this.scaType == CHIP_OTP) { // switch is better
  buildChipOtp();
} else if ( this.scaType == PHOTO_OTP) { // switch is better
  buildPhotoOtp();
} else {
  buildEmailOrSmsOtp();
}
```

Reply:
Ok. Should the tanType be built directly instead of creating one method for each type?

Reply:
Methods are cleaner and easier to extend, you can select whatever approach you want